ls1intum/Artemis
# Feature/jenkins templates

### Checklist

### Motivation and Context
This pull request was created for #1303. 
No buildplan configurations existed for C & Python exercises on Jenkins

### Description
- Added a configuration file for python exercises (jenkins)
- Added a configuration file for c exercises (jenkins)
- Modified the c exercise template due to: #1406
- Edited the build scripts for the new template (bamboo)
- Updated the Jenkins dockerfile to include the required dependencies

### Steps for Testing
##Python Exercise (Jenkins)
1. Log in to https://artemistest2.ase.in.tum.de/
2. Navigate to Course Administration
3. Create a Python programming exercise
4. In the exercise details, check the solution results. 
 - All tests should have passed.
5. Start the exercise and commit the solution. 
 - All tests should pass for the exercise. 

##C Exercise (Jenkins)
1. Log in to https://artemistest2.ase.in.tum.de/
2. Navigate to Course Administration
3. Create a C programming exercise
4. In the exercise details, check the solution results. 
 - All tests should have passed.
5. Start the exercise and commit the solution. 
 - All tests should pass for the exercise. 

##C Exercise (Bamboo)
1. Log in to https://artemistest.ase.in.tum.de/
2. Navigate to Course Administration
3. Create a C programming exercise
4. In the exercise details, check the solution results. 
 - All tests should have passed.
5. Start the exercise and commit the solution. 
 - All tests should pass for the exercise. 

### Screenshots
C Exercise (jenkins)
![image](https://user-images.githubusercontent.com/15110960/81670544-1343fb80-9448-11ea-9276-3340a3a7e31b.png)

Python Exercise (jenkins)
![image](https://user-images.githubusercontent.com/15110960/81670444-e98ad480-9447-11ea-96fe-6882def2da5e.png)

C Exercise (bamboo)
![image](https://user-images.githubusercontent.com/15110960/81670655-3bcbf580-9448-11ea-854d-f30cf4c6cafb.png)
